django/views/generic/base.py

```diff
@@ -148,7 +148,16 @@ def http_method_not_allowed(self, request, *args, **kwargs):
             request.path,
             extra={"status_code": 405, "request": request},
         )
-        return HttpResponseNotAllowed(self._allowed_methods())
+        response = HttpResponseNotAllowed(self._allowed_methods())
+
+        if self.view_is_async:
+
+            async def func():
+                return response
+
+            return func()
+        else:
+            return response
```

Comment:
For `main` we could maybe pull this fragment into a decorator to be applied both here and in `options()`. 🤔

Reply:
Hi Carlton,

Sorry for the late answer. Your idea is interesting, but I often find pretty difficult to understand (and maintain) decorators definitions, in particular when you want to deal with both syntaxes `@decorated` and `@decorated()`.

In such case, I would keep this as it and create a decorator to factorize the if/else statement only if it is needed in 1 or 2 more locations.

What do you think?